core/err.go

```diff
@@ -0,0 +1,91 @@
+package core
+
+const (
+	errCodeNoErr = iota
+	errCodeBufferExpired
+	errCodeBufferTooSmall
+	errCodeCofferExpired
+	errCodeDecryptionFailed
+	errCodeInvalidKeyLength
+	errCodeNullBuffer
+	errCodeNullEnclave
+
+	// Error string context.
+	ctx = "memguard/core: "
+)
+
+type errCode uint
+
+type memguardErr struct {
+	s string
+	c errCode
+}
+
+func (e memguardErr) Error() string {
+	return e.s
+}
+
+var errors = [...]memguardErr {
```

Comment:
Please add tests for this functionality.

1. Verify that indexing `errors` by the enum declared above returns the correct error.
2. Verify that each function/method in this file does what it is meant to do.